### PR TITLE
CMake script improvements.

### PR DESCRIPTION
**Pull request by moien007**
Added two CMake changes.
- Option to select library type (shared or static)
- Linking against `ws2_32` and `winmm` if compiler is MSVC

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -84,11 +91,11 @@ set(SOURCE_FILES
...
 source_group(source FILES ${SOURCE_FILES})
 
-add_library(enet STATIC
+add_library(enet ${ENET_BUILD_LIBTYPE}
     ${INCLUDE_FILES}
     ${SOURCE_FILES}
 )
 
-if (MINGW)
+if (MINGW OR MSVC)
```
Comment (Croydon) [inline CMakeLists.txt:99]:
```suggestion
if (WIN32)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -18,8 +18,15 @@ check_struct_has_member("struct msghdr" "msg_flags" "sys/types.h;sys/socket.h" H
 set(CMAKE_EXTRA_INCLUDE_FILES "sys/types.h" "sys/socket.h")
 check_type_size("socklen_t" HAS_SOCKLEN_T BUILTIN_TYPES_ONLY)
 unset(CMAKE_EXTRA_INCLUDE_FILES)
+
+set(ENET_BUILD_LIBTYPE "STATIC" CACHE STRING "ENet library build type. (SHARED/STATIC)")
```
Comment (Croydon) [inline CMakeLists.txt:22]: Please don't use a custom option but the standard `BUILD_SHARED_LIBS`

https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html#variable:BUILD_SHARED_LIBS

Comment (Croydon) [inline CMakeLists.txt:22]:
```suggestion
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -84,11 +91,11 @@ set(SOURCE_FILES
 source_group(include FILES ${INCLUDE_FILES})
 source_group(source FILES ${SOURCE_FILES})
 
-add_library(enet STATIC
+add_library(enet ${ENET_BUILD_LIBTYPE}
```
Comment (Croydon) [inline CMakeLists.txt:94]:
```suggestion
add_library(enet
```

Not necessary, CMake will automatically use the value of `BUILD_SHARED_LIBS`

https://cmake.org/cmake/help/latest/command/add_library.html

File: CMakeLists.txt [hunk 4]
```diff
@@ -18,8 +18,15 @@ check_struct_has_member("struct msghdr" "msg_flags" "sys/types.h;sys/socket.h" H
 set(CMAKE_EXTRA_INCLUDE_FILES "sys/types.h" "sys/socket.h")
 check_type_size("socklen_t" HAS_SOCKLEN_T BUILTIN_TYPES_ONLY)
 unset(CMAKE_EXTRA_INCLUDE_FILES)
+
+set(ENET_BUILD_LIBTYPE "STATIC" CACHE STRING "ENet library build type. (SHARED/STATIC)")
+
+if (ENET_BUILD_LIBTYPE STREQUAL "STATIC")
```
Comment (Croydon) [inline CMakeLists.txt:24]:
```suggestion
if (BUILD_SHARED_LIBS)
```

**Comment by Croydon** [comment]
Might be of interest for @spaceIm due to https://github.com/conan-io/conan-center-index/pull/2528

**Comment by bjorn** [comment]
@moien007 This pull request no longer makes sense in its current state, since it's based on your `master` which contains a large amount of unrelated changes. I'd suggest to open separate pull requests from dedicated branches, one for each suggested change.

**Comment by moien007** [comment]
@bjorn Ah, thanks for the heads up.

**Pull request closed without merging**